File: inject-java/src/test/java/io/micronaut/aop/introduction/Stub.java
package io.micronaut.aop.introduction;

import io.micronaut.aop.Introduction;
import io.micronaut.context.annotation.Type;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import static java.lang.annotation.RetentionPolicy.RUNTIME;

/**
 * Marker annotation used in tests to apply introduction advice.
 */
@Introduction
@Type(StubIntroducer.class)
@Documented
@Retention(RUNTIME)
@Target({ElementType.TYPE, ElementType.METHOD, ElementType.FIELD, ElementType.PARAMETER, ElementType.ANNOTATION_TYPE})
public @interface Stub {
}

File: test-inject-kotlin2-ksp2/src/test/kotlin/io/micronaut/kotlin/processing/aop/introduction/Stub.kt
package io.micronaut.kotlin.processing.aop.introduction

import io.micronaut.aop.Introduction
import io.micronaut.context.annotation.Type

@Introduction
@Type(StubIntroducer::class)
@MustBeDocumented
@Retention
annotation class Stub(val value: String = "")
